src/build.cc

```diff
@@ -1071,8 +1074,9 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
   } else if (deps_type == "gcc" || deps_type == "msvc_source_dependencies") {
     string depfile = result->edge->GetUnescapedDepfile();
     if (depfile.empty()) {
-      *err = string("edge with deps=") + deps_type +
-             string("but no depfile makes no sense");
+      *err = string("edge with deps=");
+      err->append(deps_type);
+      err->append("but no depfile makes no sense");
```

Comment:
space before "but". A test covering this would also be good.

Reply:
Added space and test.